### PR TITLE
[corlib] Add support for new calling convention.

### PR DESCRIPTION
**Pull request by kumpera**
As per #runtime discussion. Add support for the new calling convention optimized for awesomeness.

### CODE REVIEW

File: mcs/class/referencesource/mscorlib/system/runtime/interopservices/callingconvention.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public enum CallingConvention
         StdCall         = 3,
         ThisCall        = 4,
         FastCall        = 5,
+		Barbaric		= 9000,
```
Comment (xmcclure) [inline mcs/class/referencesource/mscorlib/system/runtime/interopservices/callingconvention.cs]: Indentation is inconsistent

**Comment by xmcclure** [comment]
Please add more comments

File: mcs/class/referencesource/mscorlib/system/runtime/interopservices/callingconvention.cs (hunk 1 above)
Comment (akoeplinger) [inline mcs/class/referencesource/mscorlib/system/runtime/interopservices/callingconvention.cs]: Additionally when modifying referencesource please always wrap your changes with #if MONO.

**Comment by evincarofautumn** [comment]
Build fails—please fix before merging.

**Pull request closed without merging**